omod/src/main/java/org/openmrs/module/appui/UiSessionContext.java

```diff
@@ -46,6 +48,33 @@ public UiSessionContext() {
 
     }
 
+    public String getViewProvidersPrivilege()
+    {
+    	PrivilegeConstants privilegeConstants = new PrivilegeConstants();
+    	String viewProvidersPrivilege = "";
+    	try {
+    		// openmrsVersion 1.9
+    		Class<?> privilegeConstantsClass = privilegeConstants.getClass();
+    		Field viewProvidersPrivilegeConstant = privilegeConstantsClass.getDeclaredField("VIEW_PROVIDERS");
+    		viewProvidersPrivilege = (String)viewProvidersPrivilegeConstant.get(privilegeConstants);
+    	} catch (NoSuchFieldException ex) {
+    		// do nothing
+ 		} catch (IllegalAccessException ex) {
+			// do nothing
+		}
+    	try {
+    		// openmrsVersion 2.0
+    		Class<?> privilegeConstantsClass = privilegeConstants.getClass();
+    		Field viewProvidersPrivilegeConstant = privilegeConstantsClass.getDeclaredField("GET_PROVIDERS");
+    		viewProvidersPrivilege = (String)viewProvidersPrivilegeConstant.get(privilegeConstants);
+    	} catch (NoSuchFieldException ex) {
+    		// do nothing
+ 		} catch (IllegalAccessException ex) {
+			// do nothing
+		}
+    	return viewProvidersPrivilege;
+    }
+    
```

Comment:
Just to be sure the conditional will have to be part of the build not the runtime. I know of version conditionals for runtime but not for build time except by using separate maven modules which seems to be heavy handed for just one line in one class.
I will be interested in learning more about your thoughts on a better way to approach this.